ambry-cloud/src/main/java/com/github/ambry/cloud/CloudBlobStore.java

```diff
@@ -419,32 +417,34 @@ public short undelete(MessageInfo info) throws StoreException {
    * @throws CloudStorageException in case any exception happens during undelete.
    */
   private short undeleteIfNeeded(BlobId blobId, short lifeVersion) throws CloudStorageException {
-    // TODO: Currently this is implemented for undeletes via replication only for DR.
     // See note in deleteIfNeeded.
     if (!checkCacheState(blobId.getID(), lifeVersion, BlobState.CREATED)) {
-      short newLifeVersion = cloudDestination.undeleteBlob(blobId, lifeVersion);
+      short newLifeVersion = cloudDestination.undeleteBlob(blobId, lifeVersion,
+          (metadata, key, updateFields) -> preUndeleteValidation(metadata, key, updateFields));
```

Comment:
could you simplify the lambda to `this::preUndeleteValidation`? Same in other places where this method is used.